Documentation/Evolution/StringProcessingAlgorithms.md

```diff
@@ -490,6 +574,23 @@ extension RangeReplaceableCollection where SubSequence == Substring {
         subrange: Range<Index>,
         maxReplacements: Int = .max
     ) -> Self where Replacement.Element == Element
+
+    /// Returns a new collection in which all occurrences of a sequence matching
+    /// the given regex are replaced by another collection.
+    /// - Parameters:
+    ///   - replacement: The new elements to add to the collection.
+    ///   - subrange: The range in the collection in which to search for `regex`.
+    ///   - maxReplacements: A number specifying how many occurrences of the
+    ///   sequence matching `regex` to replace. Default is `Int.max`.
+    ///   - content: A closure to produce a `RegexComponent` to replace.
+    /// - Returns: A new collection in which all occurrences of subsequence
+    /// matching `regex` in `subrange` are replaced by `replacement`.
+    public func replacing<Replacement: Collection>(
```

Comment:
I'm still not quite sure about these `replace` ones. With the simplest one that has default value for `subrange` and `maxReplacements`, you'd do 

```swift
let _ = str.replacing(with: "xxx") {
    OneOrMore("a")
    "+"
    OneOrMore("a")
}
```

But this reads backwards to me and doesn't translate great.

Reply:
Good point. But seems pretty reasonable given the syntactic benefit it offers.

Reply:
ok I'll leave them here for practicality